### ISSUE TITLE
SA1008, SA1012: false positives for property pattern inside parentheses

### ISSUE THREAD

**Issue by adiessl** [DotNetAnalyzers/StyleCopAnalyzers#3812]
Using `StyleCop.Analyzers` version `1.2.0-beta.556`

```csharp
public static class Helpers
{
    public static bool HasValidLength(string s)
    {
        // ⚠ SA1008 Opening parenthesis should not be preceded by a space.
        // ⚠ SA1012 Opening brace should be preceded by a space
        //              ↓↓
        return s is not ({ Length: < 5 } or { Length: > 10 });
    }
}
```

This is similar to the problem fixed in #3141.

**Comment by bjornhellander** [comment]
I can confirm SA1012, but I actually don't see a SA1008 in this example. Weird.

**Comment by adiessl** [comment]
Yes, technically correct it would be like this:

```csharp
public static class Helpers
{
    public static bool HasValidLength(string s)
    {
        // ⚠ SA1012 Opening brace should be preceded by a space
        //               ↓
        return s is not ({ Length: < 5 } or { Length: > 10 });
    }
}
```

"Fixing" this leads to the other error:

```csharp
public static class Helpers
{
    public static bool HasValidLength(string s)
    {
        // ⚠ SA1008 Opening parenthesis should not be preceded by a space.
        //                ↓
        return s is not ( { Length: < 5 } or { Length: > 10 });
    }
}
```

**Comment by bjornhellander** [comment]
Ok 👍

I am also seeing the same for `({ Length: < 5 } and { Length: > 2 })` and `({ Length: < 5 })`
